packages/probers/cypress/e2e/uploadTrack.cy.ts

```diff
@@ -163,35 +163,29 @@ describe('Upload', () => {
         '0'
       )
 
-      cy.waitUntil(() => {
-        return cy
-          .findByRole('progressbar', { name: /upload in progress/i })
-          .then((progressbar) => {
-            return Number(progressbar.attr('aria-valuenow')) > 0
-          })
-      })
-
-      cy.waitUntil(
-        () => {
-          return cy
-            .findByRole('progressbar', { name: /upload in progress/i })
-            .then((progressbar) => {
-              return Number(progressbar.attr('aria-valuenow')) > 50
-            })
-        },
-        { timeout: 40000, interval: 1000 }
-      )
-
-      cy.waitUntil(
-        () => {
-          return cy
-            .findByRole('progressbar', { name: /upload in progress/i })
-            .then((progressbar) => {
-              return Number(progressbar.attr('aria-valuenow')) === 100
-            })
-        },
-        { timeout: 40000, interval: 1000 }
-      )
+      const assertProgress = (progress: number) => {
+        cy.waitUntil(
+          () => {
+            return cy
+              .findByRole('progressbar', { name: /upload in progress/i })
+              .then((progressbar) => {
+                return Number(progressbar.attr('aria-valuenow')) > progress
+              })
+          },
+          { timeout: 40000, interval: 1000 }
+        )
+      }
+
+      assertProgress(0)
+      assertProgress(10)
+      assertProgress(20)
+      assertProgress(30)
+      assertProgress(40)
+      assertProgress(50)
+      assertProgress(60)
+      assertProgress(70)
+      assertProgress(80)
+      assertProgress(90)
```

Comment:
lol

Reply:
do we want a check for the overall time taken? could be flaky, but might be useful

Reply:
yeah this is so bad right? i tried to do a for-loop but seems to not work with how cypress does things.

Reply:
yeah that could probably also work, too, hoping to follow the status a little closer than just some really long timeout if possible

Reply:
totally, just as a sanity check in addition
I wasn't 100% sure whether the above enforces a reasonable implicit timeout for the whole thing